### PR TITLE
Add doing it wrong if namespace is incorrectly formatted

### PR DESCRIPTION
**Pull request by TimothyBJacobs**
Trac ticket: https://core.trac.wordpress.org/ticket/49749

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**

### CODE REVIEW

File: src/wp-includes/rest-api.php [hunk 1]
```diff
@@ -44,6 +44,12 @@ function register_rest_route( $namespace, $route, $args = array(), $override = f
 		return false;
 	}
 
+	$clean_namespace = trim( $namespace, '/' );
+
+	if ( $clean_namespace !== $namespace ) {
+		_doing_it_wrong( __FUNCTION__, __( 'Namespace must not start or begin with a slash.' ), '5.4.2' );
```
Comment (todeveni) [inline src/wp-includes/rest-api.php:50]: > start or begin

Should it be `start or end`?

Comment (TimothyBJacobs) [reply src/wp-includes/rest-api.php]: That would make a lot more sense 😃. Thanks for catching that @todeveni! Fixed in e24bec1.

File: src/wp-includes/rest-api.php [hunk 2]
```diff
@@ -84,8 +90,8 @@ function register_rest_route( $namespace, $route, $args = array(), $override = f
 		$arg_group['args'] = array_merge( $common_args, $arg_group['args'] );
 	}
 
-	$full_route = '/' . trim( $namespace, '/' ) . '/' . trim( $route, '/' );
-	rest_get_server()->register_route( $namespace, $full_route, $args, $override );
+	$full_route = '/' . trim( $clean_namespace, '/' ) . '/' . trim( $route, '/' );
```
Comment (skarabeq) [inline src/wp-includes/rest-api.php:93]: I didn't understand why you trim twice the local variable `$clean_namespace`?  It is unnecessary.
Test result

Comment (TimothyBJacobs) [reply src/wp-includes/rest-api.php]: Good point! I've adjusted this in dbc648a97e.

**Pull request closed without merging**